packages/math/src/GroupD8.js

```diff
@@ -1,46 +1,87 @@
-// Your friendly neighbour https://en.wikipedia.org/wiki/Dihedral_group of order 16
+// Your friendly neighbour https://en.wikipedia.org/wiki/Dihedral_group
+//
+// This file implements the dihedral group of order 16, also called
+// of degree 8. That's why its called GroupD8.
+
 import Matrix from './Matrix';
 
+/**
+ * Maps each `PIXI.GD8Symmetry` to the direction of
+ * the X-component of the new U-axis.
+ */
 const ux = [1, 1, 0, -1, -1, -1, 0, 1, 1, 1, 0, -1, -1, -1, 0, 1];
+
+/**
+ * Maps each `PIXI.GD8Symmetry` rotation to the direction of
+ * the Y-component of the new U-axis.
+ */
 const uy = [0, 1, 1, 1, 0, -1, -1, -1, 0, 1, 1, 1, 0, -1, -1, -1];
+
+/**
+ * Maps each `PIXI.GD8Symmetry` rotation to the direction of
+ * the X-component of the new V-axis.
+ */
 const vx = [0, -1, -1, -1, 0, 1, 1, 1, 0, 1, 1, 1, 0, -1, -1, -1];
+
+/**
+ * Maps each `PIXI.GD8Symmetry` rotation to the direction of
+ * the Y-component of the new V-axis.
+ */
 const vy = [1, 1, 0, -1, -1, -1, 0, 1, -1, -1, 0, 1, 1, 1, 0, -1];
-const tempMatrices = [];
 
-const mul = [];
+/**
+ * [Cayley Table]{@link https://en.wikipedia.org/wiki/Cayley_table}
+ * for the composition of each rotation in the dihederal group D8.
+ *
+ * It is a 2D array as follows:
+ * |    | E  | SE | S  | SW | W  | NW | N  | NE |
+ * |----|----|----|----|----|----|----|----|----|
+ * | E  | E  | SE | S  | SW | W  | NW | N  | NE |
+ * | SE | SE | S  | SW | W  | NW | N  | NE | E  |
+ * | S  | S  | SW | W  | NW | N  | NE | E  | SE |
+ * | SW | SW | W  | NW | N  | NE | E  | SE | S  |
+ * | W  | W  | NW | N  | NE | E  | SE | S  | SW |
+ * | NW | NW | N  | NE | E  | SE | S  | SW | W  |
+ * | N  | N  | NE | E  | SE | S  | SW | W  | NW |
+ * | NE | NE | E  | SE | S  | SW | W  | NW | N  |
```

Comment:
I don't like big tables that are also trivial. We can fix that!

Either make it brife (just 4 elements) 
Either specify 4 main and 4 reflections - that table is not trivial. 

Also specify integer codes in a first row.

I have an idea how to make it even better but for now that'll be enough.

Reply:
I could make it only show four rotations; however, `rotationCayley` (formerly `mul`) doesn't store compositions involving reflections.

Reply:
its 16x16. It stores everything :)

Reply:
I wouldn't make this class if we it didnt support reflections. All rotations are easy, just `(x+y)%8` or `(x+8-y)%8`.

Reply:
I've done what you've said.